src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -125,10 +125,6 @@ void AbstractLQPNode::set_left_input(const std::shared_ptr<AbstractLQPNode>& lef
 }
 
 void AbstractLQPNode::set_right_input(const std::shared_ptr<AbstractLQPNode>& right) {
-  DebugAssert(right == nullptr || type == LQPNodeType::Join || type == LQPNodeType::Union ||
```

Comment:
Can you include `vector`, `memory`, `optional`, `ostream`, and `types.hpp`?

Reply:
You know that I'm not a huge fan, but I guess we agreed to use IWYU (only mild forms of violence were used though). However, do we want to include it in the guidelines then until we update the CI's tidy to enforce it?

Reply:
Luckily I removed many IWYU includes with my latest PRs, have fun